### ISSUE TITLE
Users without manage_woocommerce capability get constant 403 errors

### ISSUE THREAD

**Issue by WiredWonder** [woothemes/woocommerce#48124]
### Prerequisites

- [X] I have carried out troubleshooting steps and I believe I have found a bug.
- [X] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

I have defined a custom role to allow fulfilment staff basic access to WooCommerce orders without allowing them to change the entire configuration, see reporting etc.

This has worked great until WooCommerce 8.6.1 whereby some frontend code is hitting two APIs that return 403s when viewing the order admin list. 

```
"GET /wp-json/wc-analytics/admin/notes?page=1&per_page=25&type=error%2Cupdate&status=unactioned&_locale=user HTTP/2.0" 403 356
"GET /wp-json/wc-admin/onboarding/tasks?_locale=user HTTP/2.0" 403 508
```

The error console on the client shows:
`{"code":"woocommerce_rest_cannot_view","message":"Sorry, you cannot list resources.","data":{"status":403}}{"code":"woocommerce_rest_cannot_create","message":"Sorry, you are not allowed to retrieve onboarding tasks.","data":{"status":403}}`

Granting the role `manage_woocommerce` removes the error, but grants the user far too much permission to the backend of the site.

My questions are:

Why are these two API calls being made in the first instance? 
Why do they require `manage_woocommerce`? This is a very wide reaching capability. 
Can we change the capability check, or avoid the call if the user does not have permissions? 

### Expected behavior

I expect a low privilege role to be able to fulfil orders but not edit site configuration. 

### Actual behavior

403s tripping off my IDS.

### Steps to reproduce

1. Create a role with edit posts, edit orders
2. View Orders admin list
3. View browser console

### WordPress Environment

`
### WordPress Environment ###

WordPress address (URL): na
Site address (URL): na
WC Version: 8.6.1
REST API Version: ✔ 8.6.1
WC Blocks Version: ✔ 11.8.0-dev
Action Scheduler Version: ✔ 3.7.1
Log Directory Writable: ✔
WP Version: 6.5.3
WP Multisite: –
WP Memory Limit: 350 MB
WP Debug Mode: –
WP Cron: ✔
Language: en_AU
External object cache: ✔

### Server Environment ###

Server Info: Apache
PHP Version: 7.4.33
PHP Post Max Size: 8 MB
PHP Time Limit: 30
PHP Max Input Vars: 1000
cURL Version: 7.74.0
OpenSSL/1.1.1w

SUHOSIN Installed: –
MySQL Version: 5.5.5-10.5.23-MariaDB-0+deb11u1
Max Upload Size: 8 MB
Default Timezone is UTC: ✔
fsockopen/cURL: ✔
SoapClient: ❌ Your server does not have the SoapClient class enabled - some gateway plugins which use SOAP may not work as expected.
DOMDocument: ✔
GZip: ✔
Multibyte String: ✔
Remote Post: ✔
Remote Get: ✔

### Database ###

WC Database Version: 8.6.1
WC Database Prefix: wp_
Total Database Size: 1917.00MB
Database Data Size: 1418.17MB
Database Index Size: 498.83MB

### Security ###

Secure connection (HTTPS): ✔
Hide errors from visitors: ✔

### Settings ###

API Enabled: –
Force SSL: ✔
Currency: AUD ($)
Currency Position: left
Thousand Separator: ,
Decimal Separator: .
Number of Decimals: 2
Taxonomies: Product Types: external (external)
grouped (grouped)
simple (simple)
variable (variable)

Taxonomies: Product Visibility: exclude-from-catalog (exclude-from-catalog)
exclude-from-search (exclude-from-search)
featured (featured)
outofstock (outofstock)
rated-1 (rated-1)
rated-2 (rated-2)
rated-3 (rated-3)
rated-4 (rated-4)
rated-5 (rated-5)

Connected to Woo.com: ✔
Enforce Approved Product Download Directories: –
HPOS feature screen enabled: –
HPOS feature enabled: –
Order datastore: WC_Order_Data_Store_CPT
HPOS data sync enabled: –

### WC Pages ###

Shop base: ❌ Page not set
Cart: #6 - /cart/
Checkout: #7 - /checkout/
My account: #8 - /my-account/
Terms and conditions: #2773 - /terms-and-conditions/

### Theme ###

Name: Enfold
Version: 1.0
Author URL: http://woocommerce.com
Child Theme: ✔
Parent Theme Name: Enfold
Parent Theme Version: 5.6.10
Parent Theme Author URL: https://kriesi.at
WooCommerce Support: ✔

### Templates ###

Overrides: enfold-child/woocommerce/emails/email-order-items.php

### Admin ###

Enabled Features: activity-panels
analytics
product-block-editor
coupons
core-profiler
customer-effort-score-tracks
import-products-task
experimental-fashion-sample-products
shipping-smart-defaults
shipping-setting-tour
homescreen
marketing
mobile-app-banner
navigation
onboarding
onboarding-tasks
product-variation-management
product-virtual-downloadable
product-external-affiliate
product-grouped
product-linked
remote-inbox-notifications
remote-free-extensions
payment-gateway-suggestions
shipping-label-banner
subscriptions
store-alerts
transient-notices
woo-mobile-welcome
wc-pay-promotion
wc-pay-welcome-page

Disabled Features: customize-store
minified-js
new-product-management-experience
product-pre-publish-modal
settings
async-product-editor-category-field

Daily Cron: ✔ Next scheduled: 2024-06-05 03:05:25 +10:00
Options: ✔
Notes: 178
Onboarding: completed

### Braintree (Credit Card) ###

Environment: Production
Tokenization Enabled: ✔
Debug Mode: Off

### Braintree (PayPal) ###

Environment: Production
Tokenization Enabled: –
Debug Mode: Off

### Action Scheduler ###

Complete: 16,802
Oldest: 2024-05-04 16:22:51 +1000
Newest: 2024-06-04 15:49:49 +1000

Failed: 34
Oldest: 2023-02-02 21:36:15 +1100
Newest: 2024-04-15 10:41:57 +1000

Pending: 5
Oldest: 2024-06-04 18:17:35 +1000
Newest: 2024-06-05 02:32:56 +1000

### Status report information ###

Generated at: 2024-06-04 16:07:44 +10:00
`

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/products/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

**Comment by adrianduffell** [comment]
Thanks for reporting this, @WiredWonder . I am able to reproduce the 403 errors by creating a role with the following permissions using [user role editor](https://wordpress.org/plugins/user-role-editor/):

`edit_others_shop_orders`, `edit_posts`, `edit_shop_order`, `edit_shop_orders`, `read`, `read_shop_order`.

> Why are these two API calls being made in the first instance?

I believe the calls are being made to support the activity panel in the toolbar.

![activity panel error](https://github.com/woocommerce/woocommerce/assets/9312929/fa00e05d-c143-47cb-ba3c-48258c78331d)

> Why do they require manage_woocommerce? This is a very wide reaching capability.

In general, we expect Woo admins to use the [shop manager](https://woocommerce.com/document/roles-capabilities/) or admin role, and both have `manage_woocommerce` capability. There is perhaps an argument to be made that Woo should also include a role for handling orders only, without this capability. Can you provide more info about the reasons for the fulfilment role you've created?

 > Can we change the capability check, or avoid the call if the user does not have permissions?
 
 Yes, I think the best fix is to apply a capability check on the activity panel. It could be hidden when the user doesn't have `manage_woocommerce` capability since its purpose is to provide info to users that can mange the store fully.

**Comment by WiredWonder** [comment]
Thanks Adrian,

The use case is common in any scaled eCommerce business where the store manager is not the one fulfilling orders. On a site doing 50+ orders a day, it's unlikely that the staff fulfilling orders would need access to things like payment gateway settings, tax rates, shipping settings etc. Most of the time these staff are part time or casual workers or could even be temporary workers hired to fill a peak in demand. 

As such I have multiple roles created using Capability Manager for these purposes:

- A basic fulfilment role which can only edit orders (all capabilities ticked for Orders post type)
- A moderate role which allows both editing orders, categories, tags and products (most capabilities ticked for most post types, also a few woocommerce capabilities for assigning terms etc). This allows for both fulfilment and basic eCommerce merchandising or listing updates.
- An admin role who can edit shop settings (default shop_manager)

I feel like this is a good minimal security model without going crazy with many variations. 

Totally open to compromise.

**Comment by adrianduffell** [comment]
@WiredWonder, thanks for providing the info it's much appreciated.

I agree it's a good minimal security model, and I think it would benefit other stores for WooCommerce to provide a role out-of-the-box similar to your basic fulfill role. I'll see what I can do in this area but it will take time. 

In the meantime I'll prioritize applying a capability check on the activity panel. Stay tuned!

**Comment by WiredWonder** [comment]
Thank you for the fix - I look forward to testing when I schedule my next upgrade :)

**Comment by adrianduffell** [comment]
Thanks @WiredWonder, let me know if the fix needs any follow-ups. It's scheduled to be included with the upcoming 9.4 release.
